v2g_liberty.py

```diff
@@ -119,44 +184,160 @@ def disconnect_charger(self, *args, **kwargs):
+                "notify/mobile_app_" + recipient,
+                message="clear_notification",
+                data={"tag": tag}
+            )
+        except:
+            self.log(f"Could not clear notification: exception on {recipient}.")
+
+    def handle_no_new_schedule(self, error_name: str, error_state: bool):
+        """ Inform user about no schedule:
+            - invalid schedule
+            - timeouts on schedule
+            - no communication with FM
+            They can occur simulataniously/overlapping
+
+            When error_state = True of any of the errors:
+                Set immidiatly in UI
+                Notify once if remains for an hour
+            When error state = False:
+                If all errors are solved:
+                    Remove from UI immidiately
+                    If notification has been sent:
+                        Notify user the situation has been restored.
+        """
+        self.log(f"handle_no_valid_schedule called with {error_name}: {error_state}.")
+        self.no_schedule_errors[error_name] = error_state
+        any_errors = False
+        for error_name in self.no_schedule_errors:
+            if self.no_schedule_errors[error_name]:
+                any_errors = True
+                break
+
+        if any_errors:
+            # Apparently this is the first time only set "on" once to keep history clean
+            self.set_state("input_boolean.error_no_new_schedule_available", state="on")
+            if not self.no_schedule_notification_is_planned:
+                title = "No new schedules available"
+                message = f"The current schedule wil remain active." \
+                          f"Usually this problem is solved automatically in an hour or so." \
+                          f"If the schedule does not fit your needs, consider charging manually via the chargers app."
+                self.notification_timer_handle = self.run_in(
+                    self.notify_user(
+                        message     = message,
+                        title       = title,
+                        tag         = "no_new_schedule",
+                        critical    = False,
+                        send_to_all = True),
+                    60 * 60
+                )
+                self.no_schedule_notification_is_planned = True
         else:
-            self.notify(message, title=title)
+            self.set_state("input_boolean.error_no_new_schedule_available", state="off")
+            canceled_before_run = self.cancel_timer(self.notification_timer_handle)
+            if self.no_schedule_notification_is_planned and not canceled_before_run:
+                # Only send this message if the previous wa
+                title = "Scheduals available again"
+                message = f"The problems with schedules have been solved. " \
+                          f"If you've set charging via the chargers app, consider end that and use automatic charging agian."
```

Comment:
consider end -> consider to end
agian -> again

Reply:
OK

Reply:
Looks like you skipped my second suggestion.